module/DTO/questionsDTO.js

```diff
@@ -0,0 +1,521 @@
+// Dependencies.
+const { isEmpty, find, capitalize } = require("lodash");
+const { readQuestionSet, readQuestion } = require(ROOT_PATH +
+  "/generics/services/knowledge-platform");
+const { questionType } = require(MODULES_BASE_PATH +
+  "/questions/questionTemplate");
+const { criteriaTemplate, defaultCriteria } = require(MODULES_BASE_PATH +
+  "/criteria/criteriaTemplate");
+const { baseAssessment, assessmentTemplate } = require(MODULES_BASE_PATH +
+  "/assessments/assessmentTemplate");
+
+module.exports = class Transformation {
+  /**
+   * Fetches the question set hierarchy for a solution and handles evidence generation.
+   * Caches the result if not already cached.
+   *
+   * @method
+   * @name getQuestionSetHierarchy
+   * @param {Object} submissionDocumentCriterias - Criteria from the submission document to process the question set.
+   * @param {Object} solutionDocument
+   * @param {Boolean} [isPageQuestionsRequired=true] - Optional flag to determine if page questions are required.
+   * @returns {Promise<JSON>} - Resolves with the question set hierarchy and evidences or rejects with an error.
+   */
+  static getQuestionSetHierarchy(
+    submissionDocumentCriterias,
+    solutionDocument,
+    isPageQuestionsRequired = true
+  ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const referenceQuestionSetId = solutionDocument?.referenceQuestionSetId;
+
+        const cacheData = await redisCache
+          .get(referenceQuestionSetId)
+          .catch((err) => {
+            console.log("Error in getting data from redis:", err);
+          });
+
+        if (cacheData) {
+          resolve({
+            success: true,
+            message: messageConstants.apiResponses.EVIDENCE_FETCHED,
+            data: JSON.parse(cacheData),
+          });
+        } else {
+          const res = await readQuestionSet(referenceQuestionSetId).catch(
```

Comment:
@jayesh9747 follow the existing code format

Reply:
this code is not required 
 (err) => {
              console.log("Error", err?.response?.data);
              reject({
                success: false,
                message: err?.response?.data,
                data: false,
              });
            }
          );
          
          if u are following this   if (res.body.responseCode !== httpStatusCode.ok.code) {
            return reject({
              message: res.params.errmsg,
              status: httpStatusCode.bad_request.status,
            })
          }